src/eei.cpp

```diff
@@ -177,8 +178,10 @@ inline int64_t maxCallGas(int64_t gas) {
       heraAssert(sp <= (1024 * stackItemSize), "EVM stack pointer out of bounds.");
       heraAssert(opcode >= 0x00 && opcode <= 0xff, "Invalid EVM instruction.");
 
-      auto it = evmInstructionNames.find(static_cast<uint8_t>(opcode));
-      string opName = (it != evmInstructionNames.end()) ? it->second : "UNKNOWN";
+      const char* const* const opNamesTable = evmc_get_instruction_names_table(EVMC_BYZANTIUM);
```

Comment:
OMG, that looks hideous

Reply:
True. If you want I can change it to `const char* const*` :D

Reply:
No, because you cannot return an array in C. It must be a const pointer to an item (which is `const char*`).

Reply:
Here you can just use `auto` or `auto*`.

Reply:
Does that ensure constness though?

Reply:
Yes. The best would be `const auto`.